### ISSUE TITLE
Collect (type up?) some useful ML resources for economists (ugrad/phd/.....)

### ISSUE THREAD

**Issue by ArnavSood** [ubcecon/computing_and_datascience#30]
Some ideas: 

- *Overview*: Classification/Clustering. Bayes classifier. Hypothesis classes. Some economic applications. Notion of _margin_. "Sample complexity." *FEATURE SELECTION.* (could be a whole course, honestly). Some inequalities (Chebyshev, Hoeffding, Azuma, etc.).

- *"Overfitting"/Model Complexity*: Go over common "thumb-based" ways of assessing overfitting (degree of a polynomial, # of free parameters, BIC/AIC, etc.). Define VC-dimension, and explain why it's better (elegant, unified way to talk about complexity of _any_ hypothesis set). Give some examples. Define generalization error. Derive a generalization bound for a simple algorithm, and show how the VC-term and the empirical error (sample error) term trade off. Explain how this is a better way to think about overfitting (at least in theory). 

- *Distribution Estimation*: Maximum entropy models (define entropy, use it to set up different distance measures for distributions like Bregman). Kolmogorov probably names something. Can put some theory in here- metric spaces, etc. Do regularization, conditioning. Some Bayesian stuff would be good too (*TODO: read some stuff about Bayesian ML*). 

- *Gradient Descent Methods*: Basic idea. Plus variants- stochastic gradient, projected gradient, mirror descent, exponential. *Beauty of mirror descent- don't need gradient information directly!* Convex/nice functions. Some heuristics (learning rate?) Doubling trick. 

- *Boosting*: Basic idea- combine a bunch of weak learners, get a strong learner. AdaBoost. Maybe good papers: [Ng at Columbia](http://www.columbia.edu/~sn2294/papers/recession.pdf), ETC. Random forests/bagging. (*TODO: read more on boosting/bagging, especially empirics*).

- *Kernels*: Basic idea- implicitly define similarity function. Can do some theory (RKHS, etc.). Motivation- lets us use "linear" algorithms (like SVM) for "nonlinear" problems. Show kernelized versions of some algorithms, with empirics. 

- *Learning and Games*: Define some online learning stuff (swap/internal/external regret). Show that regret minimization gives us a correlated equilibrium. Do proof of Nash theorem using maximum entropy models (e.g., problem 3 from [link](https://cs.nyu.edu/~mohri/aml18/hw1.pdf)). Some neat papers (e.g., copying algorithms for ZSG- [link](http://cs.tau.ac.il/~mfeldman/papers/FKTc10.pdf)). Define _learning equilibrium_, and put some stuff on it (*TODO*: read more on learning equilibrium). 

Ref: 

- Course at NYU [link](https://cs.nyu.edu/cs/faculty/mohri/)
